cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -94,29 +87,20 @@ func (cm *certManagerClient) Images() ([]string, error) {
 	return images, nil
 }
 
-// EnsureWebhook makes sure the cert-manager Web-hook is Available in a cluster:
+// EnsureCertManager makes sure the cert-manager Web-hook is Available in a cluster:
 // this is a requirement to install a new provider
 // Nb. In order to provide a simpler out-of-the box experience, the cert-manager manifest
 // is embedded in the clusterctl binary.
-func (cm *certManagerClient) EnsureWebhook() error {
+func (cm *certManagerClient) EnsureCertManager() error {
 	log := logf.Log
 
-	// Checks if the cert-manager web-hook already exists, if yes, exit immediately
```

Comment:
I've re-added this check with an additional function that calls Get on each resource in the cluster to verify whether cert-manager resources already exist before Creating them all again.

Now that I think some more though, it _may_ be better to just call Create(), because currently if a single IsNotFound error is encountered, we go ahead and call Create anyway.

Whilst we may not want to address a proper upgrade process in this PR, because it _is_ upgrading the manifest used, we probably want to implement some guards to avoid users downloading the new `clusterctl` and running `init` against an existing management cluster, as they could end out with orphaned resources from the previous release leftover.

I'm actually not 100% certain on how best to address this - updating the manifests in place *without* a defined upgrade process is tough, because we don't have any means to understand whether an older version _is_ installed (unless we retain both copies of the cert-manager install manifests and explicitly check if all resources in an older version exist, but that gets.. slippery).

We _do_ have this annotation on resources: `helm.sh/chart: 'cert-manager-v0.15.2'` but I'm not sure if relying on the cert-manager project to always set this in this format is wise. Perhaps we could annotate created objects (similar to what's done already), but with a version number too? For the 'transitionary period' for users coming from earlier clusterctl versions, we could check for the presence of the `cert-manager` value with _no_ version annotation in the `clusterctl.cluster.x-k8s.io/core` annotation). But without recording exactly what resources are required to be installed in earlier versions of cert-manager, it won't be possible for us to be certain *all* resources exist (i.e. if a previous invocation of `clusterctl init` failed half way through, we'll struggle to recover).